app/scripts/components/home/featured-stories.tsx

```diff
@@ -87,10 +87,12 @@ function FeaturedStories() {
                     title={d.name}
                     tagLabels={[getString('stories').one]}
                     parentTo={STORIES_PATH}
-                    description={i === 0 ? d.description : undefined}
+                    description={
+                      i === 0 ? d.cardDescription ?? d.description : undefined
+                    }
                     date={d.pubDate ? new Date(d.pubDate) : undefined}
-                    imgSrc={d.media?.src}
-                    imgAlt={d.media?.alt}
+                    imgSrc={d.cardMedia?.src ?? d.media?.src}
+                    imgAlt={d.cardMedia?.alt ?? d.media?.alt}
```

Comment:
The logic here is a bit hard to read. Maybe we can reuse the getDescription functions above, to remove a few `??` `:` `?`

Reply:
Thanks, I now re-factored the utils which also got me thinking about the testing part, so I added a bunch of unit tests that cover all the possible cases I could think of.